xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-toc/src/main/java/org/xwiki/rendering/internal/macro/toc/XWikiTocMacro.java

```diff
@@ -39,11 +47,28 @@
 @Singleton
 public class XWikiTocMacro extends AbstractTocMacro<XWikiTocMacroParameters>
 {
+
+    /**
+     * The HTML class attribute name.
+     */
+    protected static final String CLASS = "class";
+
     /**
      * Create and initialize the descriptor of the macro.
      */
     public XWikiTocMacro()
     {
         super(XWikiTocMacroParameters.class);
     }
+
+    @Override
+    public List<Block> execute(XWikiTocMacroParameters parameters, String content, MacroTransformationContext context)
+        throws MacroExecutionException
+    {
+        List<Block> blocks = super.execute(parameters, content, context);
+        GroupBlock wrapper = new GroupBlock();
+        wrapper.setParameter(CLASS, "xtoc");
```

Comment:
Could be nice to have a comment to explain why we need a GroupBlock + class attribute here.

Reply:
Thanks @vmassol. I'm adding a link to the [discussion](https://forum.xwiki.org/t/improving-the-html-structure-of-toc-macro-output/10534/4] where `xtoc` was proposed by @lucaa. I see that the LiveData component uses CSS class `xwiki-livedata`. We have also `xwiki-selectize`... What about `xwiki-toc`?  (it seems the  `x` prefix is mostly for forms)

Reply:
Yes, this would be a new API.

Reply:
XWikiTocMacro is located in the generic package `org.xwiki.rendering.internal.macro.toc` but in an xwiki-platform project `xwiki-platform-rendering-macros` so I'm not sure if it is expected to be used within XWiki only or it may be used outside XWiki. I'm asking because this has an impact on the class name we choose. If XWiki only we could go for `xwiki-toc`. Otherwise we may consider `wikitoc`, taking example on the classes used for links (`wikilink`, `wikicreatelink`) etc.  Cc @lucaa with whom I discussed the topic separately (thanks). Wdyt?

Reply:
I've added a comment on why a GroupBlock is needed with a class attribute, using `xwiki-toc` since my understanding is that XWikiTocMacro is meant to be used in XS only (and several XWiki DOM elements such LiveData use the CSS class `xwiki-livedata` naming scheme), as opposed to TocMacro which is generic.